lang/src/org/partiql/lang/eval/visitors/OrderByAliasVisitorTransform.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package org.partiql.lang.eval.visitors
+
+import org.partiql.lang.ast.IsTransformedOrderByAliasMeta
+import org.partiql.lang.domains.PartiqlAst
+import org.partiql.lang.domains.metaContainerOf
+import org.partiql.pig.runtime.SymbolPrimitive
+
+/**
+ * A [PartiqlAst.VisitorTransform] to replace the [PartiqlAst.SortSpec] of a [PartiqlAst.OrderBy] with a reference to
+ * a [PartiqlAst.ProjectItem]'s [PartiqlAst.Expr] if an alias is provided. Also utilizes [IsTransformedOrderByAliasMeta]
+ * to enforce idempotency (delivering the same result through multiple passes).
+ *
+ * Turns:
+ *
+ * ```SELECT a + 1 AS b FROM c ORDER BY b```
+ *
+ * Into:
+ *
+ * ```SELECT a + 1 AS b FROM c ORDER BY a + 1```
+ */
+class OrderByAliasVisitorTransform : VisitorTransformBase() {
```

Comment:
nit: The name `OrderByAlias` can be somewhat confusing, i.e., one may think of it as `ORDER BY <expr> AS <id>`
Maybe change it to `OrderBySortSpecVisitorTransform`? Since we essentially want to rewrite the SortSpec. 
```suggestion
class OrderBySortSpecVisitorTransform : VisitorTransformBase() {
```